src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -181,33 +199,23 @@ Pathfinder::Pathfinder (
         mDstAmount (saDstAmount),
         mSrcCurrency (uSrcCurrency),
         mSrcIssuer (uSrcIssuer),
-        mSrcAmount ({uSrcCurrency, uSrcIssuer}, 1u, 0, true),
+        convert_all_ (mDstAmount ==
+            STAmount (mDstAmount.issue (), 1u, 0, true)),
```

Comment:
No, _please_ don't muck around with the internal representation of `STAmount` in this way - the mantissa and exponent are exposed today but won't be tomorrow. Also, you aren't taking into account the fact that the value gets normalized, so the mantissa and exponent may not have the values you specify unless you're dealing with XRP.

The way it is now is fine, but if you're concerned about performance, simply construct a static `STAmount` to the value you want and use it to initialize with or compare against or whathaveyou.